src/ArcCommander/APIs/AssayAPI.fs

```diff
@@ -691,327 +692,231 @@ module AssayAPI =
-            //        match study.Contacts with
-            //        | Some persons -> 
-            //            if API.Person.existsByFullName firstName midInitials lastName persons then
-            //                API.Person.updateByFullName updateOption person persons
-            //                |> API.Assay.setContacts study
-
-            //            else
-            //                if verbosity >= 1 then printfn "Person with the name %s %s %s does not exist in the study with the identifier %s" firstName midInitials lastName studyIdentifier
-            //                if containsFlag "AddIfMissing" personArgs then
-            //                    if verbosity >= 1 then printfn "Registering person as AddIfMissing Flag was set" 
-            //                    API.Person.add persons person
-            //                    |> API.Assay.setContacts study
-            //                else 
-            //                    if verbosity >= 2 then printfn "AddIfMissing argument can be used to register person with the update command if it is missing" 
-            //                    study
-            //        | None -> 
-            //            if verbosity >= 1 then printfn "The study with the identifier %s does not contain any persons" studyIdentifier
-            //            if containsFlag "AddIfMissing" personArgs then
-            //                if verbosity >= 1 then printfn "Registering person as AddIfMissing Flag was set" 
-            //                [person]
-            //                |> API.Assay.setContacts study
-            //            else 
-            //                if verbosity >= 2 then printfn "AddIfMissing argument can be used to register person with the update command if it is missing" 
-            //                study
-            //        |> fun s -> API.Assay.updateByIdentifier API.Update.UpdateAll s studies
-            //        |> API.Investigation.setAssays investigation
-            //    | None -> 
-            //        if verbosity >= 1 then printfn "Study with the identifier %s does not exist in the investigation file" studyIdentifier
-            //        investigation
-            //| None -> 
-            //    if verbosity >= 1 then printfn "The investigation does not contain any studies"  
-            //    investigation
-            //|> Investigation.toFile investigationFilePath
-        
-        /// Opens an existing person by fullname (lastName, firstName, MidInitials) in the ARC investigation study with the text editor set in globalArgs.
-        let edit (arcConfiguration : ArcConfiguration) (personArgs : Map<string,Argument>) =
+            let persons = MetaData.getPersons "Investigation" doc
+
+            if API.Person.existsByFullName firstName midInitials lastName persons then
+                API.Person.updateByFullName updateOption person persons
+                |> MetaData.overwriteWithPersons "Investigation" <| doc
```

Comment:
I think backward piping (BWP) is rarely used and therefore might decrease readability for most users. Maybe you could bind the above value first